plugin-server/src/utils/utils.ts

```diff
@@ -698,3 +699,28 @@ export function intToBase(num: number, base: number): string {
 export class RaceConditionError extends Error {
     name = 'RaceConditionError'
 }
+
+export interface StalenessCheckResult {
+    isServerStale: boolean
+    timeSinceLastActivity: number | null
+    lastActivity?: string | null
+    lastActivityType?: string
+    instanceId?: string
+}
+
+export function stalenessCheck(hub: Hub | undefined, stalenessSeconds: number): StalenessCheckResult {
+    let isServerStale = false
+
+    const timeSinceLastActivity = hub?.lastActivity ? new Date().valueOf() - hub.lastActivity : null
+    if (timeSinceLastActivity && timeSinceLastActivity > stalenessSeconds * 1000) {
```

Comment:
what if I'm running an installation that runs events only once a week?

Reply:
More context: the staleness check was added as a debugging tool early on in the plugin server, when we noticed some pods would randomly stall, and this was a workaround to avoid always being on call to restart them.  Each pod was failing randomly within 3-24h of being started, so eventually with 3 pods we'd stop ingesting. Those issues have since been fixed, and we never removed this check due to caution and amnesia.